### PR TITLE
[Function] Support bitand, bitor, bitxor, bitnot function on vectorized engine

### PR DESCRIPTION
**Pull request by zenoyang**
## Proposed changes

Support bitand, bitor, bitxor, bitnot function on vectorized engine

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/simple_function_factory.h [hunk 1]
```diff
@@ -36,6 +36,10 @@ void register_function_minus(SimpleFunctionFactory& factory);
 void register_function_multiply(SimpleFunctionFactory& factory);
 void register_function_divide(SimpleFunctionFactory& factory);
 void register_function_int_div(SimpleFunctionFactory& factory);
+void register_function_bitand(SimpleFunctionFactory& factory);
```
Comment (zbtzbtzbt) [inline be/src/vec/functions/simple_function_factory.h:39]: 为什么不都放bitmap.cpp这里面呢？[here](https://github.com/doris-vectorized/doris-vectorized/blob/0ba567c7256243d278685c716e8d35d872f456dd/be/src/vec/functions/simple_function_factory.h#L127)

Comment (zenoyang) [reply be/src/vec/functions/simple_function_factory.h:39]: 这是位运算代码，和bitmap不是一个东西 : )

File: be/src/vec/functions/bitnot.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+// specific language governing permissions and limitations
+// under the License.
+
+#include "vec/data_types/number_traits.h"
+#include "vec/functions/function_unary_arithmetic.h"
+#include "vec/functions/simple_function_factory.h"
+
+namespace doris::vectorized {
+
+template <typename A>
```
Comment (zbtzbtzbt) [inline be/src/vec/functions/bitnot.cpp:24]: `A`这个类型我感觉写`T`好些，前面的`A` `B`也是.

File: be/src/vec/functions/simple_function_factory.h (hunk 1 above)
Comment (zbtzbtzbt) [reply be/src/vec/functions/simple_function_factory.h:39]: 哦哦了解，我觉得可以都放类似这样一个文件里：`function_binary_arithmetic.cpp`

Comment (zenoyang) [reply be/src/vec/functions/simple_function_factory.h]: 嗯嗯，有道理，已处理

File: be/src/vec/functions/function_bit.cpp [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    static constexpr auto name = "bitnot";
+};
+
+template <typename A>
+struct BitNotImpl {
+    using ResultType = typename NumberTraits::ResultOfBitNot<A>::Type;
+
+    static inline ResultType apply(A a) { return ~static_cast<ResultType>(a); }
+
+#if USE_EMBEDDED_COMPILER
```
Comment (HappenLee) [inline be/src/vec/functions/function_bit.cpp:60]: 这个代码不要引入了

**Pull request merged**